source/host_resolver.c

```diff
@@ -24,14 +24,14 @@ int aws_host_address_copy(const struct aws_host_address *from, struct aws_host_a
     to->address = aws_string_new_from_string(to->allocator, from->address);
 
     if (!to->address) {
-        return AWS_OP_ERR;
+        return aws_raise_error(AWS_IO_DNS_QUERY_FAILED);
     }
 
     to->host = aws_string_new_from_string(to->allocator, from->host);
 
     if (!to->host) {
         aws_string_destroy((void *)to->address);
-        return AWS_OP_ERR;
+        return aws_raise_error(AWS_IO_DNS_QUERY_FAILED);
```

Comment:
either undo these changes, or delete these lines entirely.

If these were to fail, it would be due to AWS_ERROR_OOM (out of memory). Reporting AWS_IO_DNS_QUERY_FAILED instead is a much less useful error for the user.

On the other hand, it's no longer possible for aws_string_new_from_string() to fail, because we no longer allow memory allocation to fail.

Reply:
Thanks, fixed.